File: backend/internal/dto/log.go
package dto

import "time"

type Log struct {
	ID                 int           `json:"id"`
	Name               string        `json:"name"`
	Message            string        `json:"message"`
	Service            string        `json:"service"`
	AppVersion         string        `json:"app_version"`
	Platform           string        `json:"platform"`
	Type               string        `json:"type"`
	APIRequests        int           `json:"api_requests"`
	SuccessRatePercent float64       `json:"success_rate_percent"`
	TrackedAt          time.Time     `json:"tracked_at"`
	AvgResponseTime    time.Duration `json:"avg_response_time"`
	ElapsedTime        time.Duration `json:"elapsed_time"`
	TimeSinceLastTrack time.Duration `json:"time_since_last_track"`
}

type LogsPaged struct {
	Logs        []Log `json:"logs"`
	CurrentPage int   `json:"current_page"`
	Pages       int   `json:"pages"`
}

File: frontend/src/interfaces/Log.ts
interface Log {
    id?: number;
    name?: string;
    message?: string;
    service?: string;
    app_version?: string;
    platform?: string;
    type?: string;
    api_requests?: number;
    success_rate_percent?: number;
    tracked_at: string; // ISO 8601 string from time.Time
    avg_response_time?: number; // nanos from time.Duration
    elapsed_time?: number; // nanos from time.Duration
    time_since_last_track?: number; // nanos from time.Duration
}